src/components/LicenseAmendmentList/LicenseAmendmentList.js

```diff
@@ -26,6 +27,46 @@ export default class LicenseAmendmentList extends React.Component {
       id: PropTypes.string.isRequired,
     }),
     renderStatuses: PropTypes.bool,
+    renderWarnings: PropTypes.bool,
+    renderNotes: PropTypes.bool,
+  }
+
+  renderStatusMismatchWarnings(amendment) {
```

Comment:
This is pretty cluttered and can be cleaned up. Some examples:
- Initing everything to `null` isn't necessary. Not only are you not comparing to `null`, `null` sets of warnings in the heads of JS devs like me since you only use it if you absolutely want _null_ and not `undefined` or `false` or `0` or `""`.
  - Use `get()` for multi-level indexing like `const status = get(amendment, 'status.value')`.
  - Use destructuring for simple ones like `const { startDate, endDate } = amendment`
- The explicit `return null`s really clutter things up here. It makes it really hard to track what's actually going on and what's necessary and what isn't. Consider something like the following:

```
if (agreementStatus === statuses.CURRENT) {
  if (licenseStatus === statuses.EXPIRED || licenseStatus === statuses.REJECTED) {
    return <FormattedMessage />;
  } else if (startDate && new Date(amendment.startDate).getTime() > new Date().getTime()) {
    return <FormattedMessage />;
  } else if (endDate && new Date(amendment.endDate).getTime() < new Date().getTime()) {
    return <FormattedMessage />;
  }
}

return null;
```
- Finally, the variable names are really confusing. `licenseStatus` is not the status of the license and `agreementStatus` is not the status of the agreement.

Reply:
Also...I thought this logic seemed familiar and it looks like [we already do this kind of thing here](https://github.com/folio-org/ui-erm/blob/5cda051fa068dcf7881338507f976d74f5ac4f98/src/components/LicensesFieldArray/AmendmentsFieldArray.js#L115). Is there a way we could just generalise the validation, put it in `utilities/validators.js`, and call that validator both in that spot and in here?

Reply:
Thank you for reminding me of the neater way to do that kind of thing, I hated the way it looked with all those `else { return.. }` bits, but couldn't get my head around it, I'll definitely refactor that. As for the generalisation, I definitely can do that, but would validators be the right place for it? It's kind of it's own thing, warnings. In addition, talking to Gill it sounded like there is a possibility that the list of places we need warnings for might expand in future, so it might be good to have a separate file in `utilities` just for warnings at some point? For now I've just put them in validators, but it might be worth keeping in mind.

Reply:
I guess `validators` is a bit of a prescriptive term to use in the file name. Really, the only thing that makes them a true final-form validator is that you assign them to the `validator` prop. If you use them in the context of a warning (as I believe we do in AmendmentsFieldArray), then yeah, they're not really functioning as a truuuue final-form validator. I just called it `validators.js` because they follow the RFF pattern, but you definitely don't **have** to use it as a way of permitting/denying form submission.

That's all kind of hand-wavy...I've rewritten that above paragraph a few times so lemme know if it's not clear.

Reply:
Yeah I see your point, I think you're right. Especially since it doesn't have the same function signature as a traditional validator. I'm not against a `utilities/getAmendmentStatusConflicts.js` or something instead.